### PR TITLE
Momentum design svg icons

### PR DESCRIPTION
**Pull request by NotNestor**
## Description
* Update icons to use momentum desing svg icons if congifured
* update to allow icon set selection momentumUI (legacy font), momentumDesign (new svg based), preferMomentumDesign (try to map icons in momentumUI to momentumDesign counter part)
* update components to use new icons
* component updates when using new icons

## Screenshots:
**Before** (If applicable):

**After**:

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: web-components/webpack.config.ts [hunk 1]
```diff
@@ -232,19 +239,18 @@ const commonDist = merge(common, {
...
       trimNameEnd: 6
     }),
     new CopyWebpackPlugin([
-      { from: `${pDesign}/dist/fonts`, to: "assets/fonts" },
-      { from: `${pDesign}/dist/fonts`, to: "assets/icons/fonts" },
-      { from: `${pDesign}/dist/data/MomentumFontIcons.css`, to: "assets/styles" },
-      { from: `${pDesign}/dist/data/MomentumFontIcons.min.css`, to: "assets/styles" },
-      { from: `${pDesign}/dist/fonts`, to: "assets/fonts" },
-      { from: `${pDesign}/dist/data/scss/`, to: "assets/styles" },
       { from: `${pMomentum}/core/fonts`, to: "assets/fonts" },
```
Comment (NotNestor) [inline web-components/webpack.config.ts:254]: remove design fonts now that the svg icons are being used

File: web-components/webpack.config.ts [hunk 2]
```diff
@@ -261,12 +267,12 @@ const commonDist = merge(common, {
         test: [
           {
             folder: "./dist/types",
-            method: p => new RegExp(/\.test\.d\.ts(\.map)*$/).test(p),
+            method: (p) => new RegExp(/\.test\.d\.ts(\.map)*$/).test(p),
             recursive: true
```
Comment (NotNestor) [inline web-components/webpack.config.ts:271]: upgraded linters causing some changes

File: web-components/.prettierrc [hunk 3]
```diff
@@ -1,5 +1,6 @@
 {
   "printWidth": 120,
   "tabWidth": 2,
-  "semi": true
+  "semi": true,
+  "trailingComma": "none"
 }
```
Comment (NotNestor) [inline web-components/.prettierrc:6]: after updating prettier dependency it is now adding trailing commas by default, configure to remove this

File: web-components/.storybook/manager.ts [hunk 4]
```diff
@@ -10,5 +10,7 @@ addons.setConfig({
   isToolshown: true,
   selectedPanel: undefined,
   initialActive: 'sidebar',
-  showRoots: true,
+  sidebar: {
+    showRoots: true
+  }
 });
```
Comment (NotNestor) [inline web-components/.storybook/manager.ts:16]: fix storybook deprecation warning

File: web-components/.storybook/preview.ts [hunk 5]
```diff
@@ -1,4 +1,4 @@
-import { addParameters } from "@storybook/client-api";
+import { Parameters } from "@storybook/client-api";
 import { setCustomElements } from "@storybook/web-components";
```
Comment (NotNestor) [inline web-components/.storybook/preview.ts:2]: fix deprecation warning

File: web-components/jest.config.js [hunk 6]
```diff
@@ -1,5 +1,6 @@
 /* eslint-disable */
 module.exports = {
+  testEnvironment: "jsdom",
   collectCoverage: true,
```
Comment (NotNestor) [inline web-components/jest.config.js:4]: warning that this will be required in next version of jest

File: web-components/package.json [hunk 7]
```diff
@@ -55,24 +55,24 @@
...
     "@interactjs/actions": "1.10.3",
     "@interactjs/auto-start": "1.10.3",
+    "@interactjs/core": "1.10.3",
     "@interactjs/interact": "1.10.3",
     "@interactjs/modifiers": "1.10.3",
     "@interactjs/utils": "1.10.3",
-    "@interactjs/core": "1.10.3",
+    "@momentum-design/icons": "0.0.154",
+    "@momentum-design/tokens": "0.0.73",
     "@popperjs/core": "^2.4.4",
```
Comment (NotNestor) [inline web-components/package.json:64]: various dependency updates. updating via cli cuased re ordering

**Review by chogorma — APPROVED** [review]
One Comment.

**Pull request merged**